### ISSUE TITLE
"Saving..." indicator for the Note page.

### ISSUE THREAD

**Issue by DarthVanger** [DarthVanger/NeuralNotes#153]
After #100 is done, and we have "note page", add indicators of when the note is saved to improve UX.

**Related**: #110. We have a bug with saving note name, because it sends 2 http requests. You will probably face this bug while testing note name editor.

## Details
After user typed some text into the editor, the note should be saved. If user continues editing this note, all changes should be saved as well. So the flow is like in Google Docs: when you create a new document, a blank "unnamed" document is opened. Whatever you type in this document is saved in real time. If the user never changes the title of the document, it is saved as "Unnamed document", but the content is saved. We want a similar behavior here.

When the note is being saved, show "saving..." like in google drive:

![image](https://user-images.githubusercontent.com/6686679/96150851-75cb8700-0f13-11eb-9bbc-abe1e04668b4.png)

When the note is saved, show "saved" like in google drive:

![image](https://user-images.githubusercontent.com/6686679/96150961-93005580-0f13-11eb-8ce3-fde47e263c00.png)

If saving the note failed, show an error.

If the user tries to close the tab, before changes were saved, show the default browser's "changes on this page might not be saved, close anyway"? (for example like they do in gmail).

Don't worry about styles/design, this task is for the functionality only. Styles will be corrected later.

**Comment by DarthVanger** [comment]
Implemented, except for error handling.
Created a separate ticket for error handling: https://github.com/DarthVanger/NeuralNotes/issues/197
